### PR TITLE
Stats: Update mentions of "Site Stats" with "Jetpack Stats"

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:

Replaces "Site Stats" with "Jetpack Stats"

#### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p1HpG7-icd-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Apply this change to your Jetpack test site.
* Ensure appropriate mentions of "Site Search" has been replaced within wp-admin on the following pages:
  * Dashboard (At-a-glance)
  * Dashboard -> My Plan
  * Settings -> Traffic
  * Stats

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx [hunk 1]
```diff
@@ -478,7 +478,7 @@ class MyPlanBody extends React.Component {
 											src={ imagePath + '/jetpack-google-analytics.svg' }
 											className="jp-landing__plan-features-icon"
 											alt={ __(
-												'Site stats showing an evolution in traffic and engagement',
+												'Jetpack Stats showing an evolution in traffic and engagement',
```
Comment (grzegorz-cp) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx]: I'm not sure if this is the correct replacement. This block corresponds to Google Analytics and this is the `alt` text description for the block's image.

This block is visible after adding `Jetpack Security Daily` plan - required in an attempt to configure Google Analytics from [Tools/Marketing menu](https://wordpress.com/marketing/traffic/).

I would suggest making this text more generic with `Stats showing an evolution in traffic and engagement` or more specific with `Google Analytics Stats showing an evolution in traffic and engagement`.

**Review by grzegorz-cp — COMMENTED** [review]
Looks good in most places, but I'm not sure about one block (details in a comment below).

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx (hunk 1 above)
Comment (jsnmoon) [reply projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx]: 🤦 You're right. This looks like an alt text for our existing Google Analytics integration.

I'll revert this.

Comment (jsnmoon) [reply projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx]: Fixed in 8c67580.

**Review by grzegorz-cp — DISMISSED** [review]
Looks good, thank you for making the changes.

**Review by grzegorz-cp — APPROVED** [review]

**Review by anomiex — APPROVED** [review]
Looks fine from a code perspective. I see someone else has already reviewed from a requirements perspective.

**Pull request merged**